File: core/src/main/java/com/eloja/core/entity/Fisica.java
package com.eloja.core.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Data
@Entity
public class Fisica extends Pessoa{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @Column(nullable = false)
    private String nome;

    @Column(nullable = false)
    private String sobrenome;

    @Column(nullable = false)
    private String cpf;

}

File: frontend-react/src/models/Fisica.ts
import Endereco from './Endereco';
import Telefone from './Telefone';

class Fisica {
  id?: number;
  nome: string;
  sobrenome: string;
  cpf: string;
  endereco: Endereco;
  telefones: Telefone[];

  constructor(nome: string, sobrenome: string, cpf: string, endereco: Endereco, telefones: Telefone[], id?: number) {
    this.id = id;
    this.nome = nome;
    this.sobrenome = sobrenome;
    this.cpf = cpf;
    this.endereco = endereco;
    this.telefones = telefones;
  }
}

export default Fisica;
